src/file_pipe_log/pipe.rs

```diff
@@ -444,6 +451,32 @@ impl<F: FileSystem> PipeLog for DualPipes<F> {
         self.pipes[handle.id.queue as usize].read_bytes(handle)
     }
 
+    #[inline]
+    fn async_read_bytes(&self, ents_idx: &mut Vec<EntryIndex>) -> Result<Vec<Vec<u8>>> {
+        let mut blocks: Vec<FileBlockHandle> = vec![];
+        for (t, i) in ents_idx.iter().enumerate() {
+            if t == 0 || (i.entries.unwrap() != ents_idx[t - 1].entries.unwrap()) {
+                blocks.push(i.entries.unwrap());
+            }
+        }
+
+        let fs = &self.pipes[LogQueue::Append as usize].file_system;
+        let mut ctx = fs.new_async_io_context(blocks.len()).unwrap();
+
+        for block in blocks.iter_mut() {
+            match block.id.queue {
+                LogQueue::Append => {
+                    self.pipes[LogQueue::Append as usize].async_read(block, &mut ctx);
+                }
+                LogQueue::Rewrite => {
+                    self.pipes[LogQueue::Rewrite as usize].async_read(block, &mut ctx);
```

Comment:
Change the function to `async_read(ctx, buf)`. It's the common order to pass in a context.

Reply:
@tabokie Did you mean `async_read(ctx, blocks)`? Like
```
impl<F: FileSystem> PipeLog for DualPipes<F> {
  fn async_read_bytes(blocks:Vec<FileBlockHandle>) -> Result<Vec<Vec<u8>>>{
      ...
      self.pipes[LogQueue::Append].async_read(ctx,blocks);
      let bytes = fs.async_finish(ctx);
      ...
  }
}
```

Reply:
Yes